pkg/controllers/applicationset_controller.go

```diff
@@ -46,6 +47,14 @@ func (r *ApplicationSetReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 	if err := r.Get(ctx, req.NamespacedName, &applicationSetInfo); err != nil {
 		log.Info("Unable to fetch applicationSetInfo %v", err)
 
+		var generator generators.Generator
+		generator = generators.NewListGenerator()
+		for _, tmpGenerator := range applicationSetInfo.Spec.Generators {
+			if tmpGenerator.List != nil {
+				newApplications, err :=  generator.GenerateApplications(&applicationSetInfo)
```

Comment:
I think we can also pass here the tmpGenerator and don;t have do to the same loop in the generator

Reply:
Done